Cython/Compiler/Parsing.py

```diff
@@ -3248,7 +3334,10 @@ def p_fused_definition(s, pos, ctx):
 
 def p_struct_enum(s, pos, ctx):
     if s.systring == 'enum':
-        return p_c_enum_definition(s, pos, ctx)
+        if s.peek()[1] in ('class', 'struct'):
+            return p_cpp_scoped_enum_definition(s, pos, ctx)
+        else:
+            return p_c_enum_definition(s, pos, ctx)
```

Comment:
This would also benefit from a single entry point, instead of peeking ahead. You can see `class` and `struct` as modifiers here that change the final appearance of the enum, not the way it's parsed.

Reply:
As recommended, I've made it such that we have a single parsing function